helm/hpcc/templates/_warnings.tpl

```diff
@@ -128,12 +128,13 @@ Pass in dict with root and warnings
  {{- end -}}
  {{- /* Ensure that resources have been provided for all components */ -}}
  {{- /* Build on the existing items in $ctx.missingResources created previously for thor, eclccserver and eclagent */ -}}
- {{- range $ctypes := (pick .root.Values  "dali"  "dfuserver" "esp" "roxie") -}}
-  {{- range $component := $ctypes -}}
+ {{- $components := omit (ternary (omit .root.Values "sasha") .root.values .root.Values.sasha.disabled) "global" "security" "storage" "certificates" "secrets" "vaults" "bundles" "thor" "eclccserver" "eclagent" -}}
```

Comment:
Out of curiosity, why is sasha different from every other component.  Every component has a list of items, whereas sasha has named objects under it.

Reply:
The thinking was that since the Sasha component types are hard-wired, and it doesn't make any sense to have >1 of any of them, unlike other components where it is natural/expected to have multiple (in a listt), that a dictionary was more logical.
I think in retrospect, it may have been better for consistency to make list and have the sasha type as an "application" field, and enforce at generation time that it must confirm to a one of the known sasha 'applications' and there can't be >1.
We could allow for this schema - support both for a while, and deprecate the dictionary based version eventually.. 
I think I'll open a JIRA.

Reply:
I'm not convinced this omit is clearer.. it will be the same problem (and more likely I'd argue) that it will break when someone introduces a new top-level non-component label.

I think it is better as before, with an explicit list of component types (including sasha).

@ghalliday - what are your thoughts on this now?